types/join-form.tsx

```diff
@@ -1,34 +1,51 @@
 import styles from '../styles/Home.module.css'
-import { useState } from 'react'
-import Link from 'next/link'
+import { SyntheticEvent, useRef, useState } from 'react'
 import Haikunator from 'haikunator'
 
 const JoinForm = () => {
   const haikunator = new Haikunator({ defaults: { tokenLength: 0 } })
-  const [name, setName] = useState<string>(undefined)
-  const [roomName, setRoomName] = useState<string>(`room/${haikunator.haikunate()}`)
+  const roomPath = useRef<string>(`room/${haikunator.haikunate()}`)
+  const [name, setName] = useState<string>('')
```

Comment:
The initial state of the name const is an empty string, in order to use name as a controlled input element for the lifetime of the component.